backends/bmv2/psa_switch/psaSwitch.cpp

```diff
@@ -578,7 +578,65 @@ Util::IJson* ExternConverter_InternetChecksum::convertExternObject(
     UNUSED ConversionContext* ctxt, UNUSED const P4::ExternMethod* em,
     UNUSED const IR::MethodCallExpression* mc, UNUSED const IR::StatOrDecl *s,
     UNUSED const bool& emitExterns) {
-    auto primitive = mkPrimitive("InternetChecksum");
+
+    if (em->method->name != "clear"
+            && em->method->name != "get"
+            && em->method->name != "add"
+            && em->method->name != "subtract"
+            && em->method->name != "get_state"
+            && em->method->name != "set_state") {
+        modelError("Unsupported register method %1%", mc);
+        return nullptr;
+    } else if (em->method->name == "clear" && mc->arguments->size() != 0) {
+        modelError("Expected 0 arguments for %1%", mc);
+        return nullptr;
+    } else if (em->method->name == "get" && mc->arguments->size() != 1) {
+        modelError("Expected 1 arguments for %1%", mc);
+        return nullptr;
+    } else if (em->method->name == "get_state" && mc->arguments->size() != 1) {
+        modelError("Expected 1 arguments for %1%", mc);
+        return nullptr;
+    } else if (em->method->name == "set_state" && mc->arguments->size() != 1) {
+        modelError("Expected 1 arguments for %1%", mc);
+        return nullptr;
+    } else if (em->method->name == "subtract" && mc->arguments->size() != 1) {
+        modelError("Expected 1 arguments for %1%", mc);
+        return nullptr;
+    } else if (em->method->name == "add" && mc->arguments->size() != 1) {
+        modelError("Expected 1 arguments for %1%", mc);
+        return nullptr;
+    }
```

Comment:
The `return` statement in the `if` body will prevent fall through to subsequent cases. Thus, it is not necessary to use `else if`.